homeassistant/components/bitvavo/sensor.py

```diff
@@ -0,0 +1,279 @@
+    def state(self):
+        """Return the state of the sensor."""
+        return round(float(self._get_data_property("price")), 4)
+
+    @property
+    def unique_id(self):
+        """Return a unique id for the sensor."""
+        return self._unique_id
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit the value is expressed in."""
+        return self._get_data_property("quote")
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend."""
+        return CURRENCY_ICONS.get(self.unit_of_measurement, DEFAULT_COIN_ICON)
+
+    @property
+    def device_state_attributes(self):
+        """Return additional sensor state attributes."""
+        return {
+            "symbol": self._symbol,
+            "bid_price": self._get_data_property("bid"),
+            "ask_price": self._get_data_property("ask"),
+            "bid_size": self._get_data_property("bidSize"),
+            "ask_size": self._get_data_property("askSize"),
+            "currency": self._get_data_property("base"),
+            "quote_asset": self._get_data_property("quote"),
+            "unit_of_measurement": self.unit_of_measurement,
+            "source": "Bitvavo",
+        }
+
+
+class Balance(CoordinatorEntity):
+    """Implementation of the balance sensor."""
+
+    def __init__(self, coordinator, balance):
+        """Initialize the sensor."""
+        super().__init__(coordinator)
+        self._balance = balance
+
+        self._name = f"Bitvavo Balance - {self._balance}"
+        self._unique_id = f"bitvavo_balance_{self._balance})"
+
+    def _get_data_property(self, property_name):
+        """Return the property from self.coordinator.data."""
+        return self.coordinator.data[CONF_BALANCES][self._balance][property_name]
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        state = float(self._get_data_property("available")) + float(
+            self._get_data_property("inOrder")
+        )
+        return round(state, 4)
+
+    @property
+    def unique_id(self):
+        """Return a unique id for the sensor."""
+        return self._unique_id
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit the value is expressed in."""
+        return self._get_data_property("symbol")
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend."""
+        return CURRENCY_ICONS.get(self.unit_of_measurement, DEFAULT_COIN_ICON)
+
+    @property
+    def device_state_attributes(self):
+        """Return additional sensor state attributes."""
+        return {
+            "available": self._get_data_property("available"),
+            "in order": self._get_data_property("inOrder"),
+            f"{ASSET_VALUE_BASE}_value".lower(): float(
+                self._get_data_property("asset_value_in_base_asset")
+            ),
```

Comment:
Shouldn't these be separate sensors?

Reply:
This is information only belonging to this sensor, isn't practical to separate them.